hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -182,14 +182,28 @@ public abstract void preCompact(
         .withOperationField(config.allowOperationMetadataField())
         .withPartition(operation.getPartitionPath())
         .build();
-    if (!scanner.iterator().hasNext()) {
-      scanner.close();
-      return new ArrayList<>();
-    }
 
     Option<HoodieBaseFile> oldDataFileOpt =
         operation.getBaseFile(metaClient.getBasePath(), operation.getPartitionPath());
 
+    // Considering following scenario: if all log blocks in this fileSlice is rollback, it returns an empty scanner.
+    // But in this case, we need to give it a base file. Otherwise, it will lose base file in following fileSlice.
+    if (!scanner.iterator().hasNext()) {
+      if (!oldDataFileOpt.isPresent()) {
+        scanner.close();
+        return new ArrayList<>();
+      } else {
+        // TODO: we may directly rename original parquet file if there is not evolution/devolution of schema
```

Comment:
Correct me if I misunderstand you question. The reason why we try to generate a new base file here rather than end up this compaction operation is that any upsert occurs after compaction plan generated will use compaction commit time as new log file base commit time. Such a fileSlice is comprised by new log file and basefile generated by compaction. If hoodieCompactor didn't generate a basefile for this fileSlice, Filegroup will lose all data in baseFile in new and following Fileslices.

Reply:
yes, your understanding is right. 
what i meant by "do these need to be cleaned up " is, why have commented out code from L197 to L203? do you want to remove them.

Reply:
I understand the reason why you have left the comments there. we can keep it there for now.